src/release.ss

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env gxi-script
+;; -*- Gerbil -*-
+;;; Release script for Gerbil.
+;; Usage: ./release.ss [version]
+;;
+;; Use it from a fully committed git tree.
+;;
+;; If a release version is specified (contains no "-"), a new release commit is made and tagged.
+;; If a development version is specified, a new development commit is made and tagged.
+;;
+;; If no version is specified, one is autodetected:
+;; If the current commit is a development commit, a release commit will be made and tagged
+;; If the current commit is a release commit, a development commit will be made and tagged
+;; e.g. v0.12-DEV-666-g1234567 will be turned into v0.13
+;; and v0.13 will be turned into v0.13-DEV
+
+(import
+  :gerbil/gambit/ports
+  :std/format :std/misc/ports :std/misc/process :std/srfi/13 :std/srfi/14 :std/sugar :std/pregexp)
+
+;; Where the root to gerbil is.
+(def gerbil-root
+  (path-normalize (path-expand ".." (path-directory (this-source-file)))))
+
+;; Change the directory to the root of gerbil.
+(current-directory gerbil-root)
+
+;; Check that .git is there indeed
+(unless (file-exists? ".git") (error "Cannot find .git directory"))
+
+;; Path to the version file
+(def version-path
+  "src/gerbil/runtime/gx-version.scm")
+
+;; Path to the .gitignore file
+(def ignore-path
+  ".gitignore")
+
+;; Just run the process on the current console
+(def (run-git . arguments)
+  (run-process/batch ["git" . arguments]))
+
+;; Compute next development version from the current release version
+(def (next-development-version version)
+  (string-append version "-DEV"))
+
+;; Compute next release version from the current development version
+(def (next-release-version version)
+  (let* ((dash-position (string-index version #\-))
+         (bare-version (if dash-position (substring version 0 dash-position) version))
+         (dot-position (string-index-right bare-version #\.))
+         (prefix (substring bare-version 0 (+ dot-position 1)))
+         (last-component (substring bare-version (+ dot-position 1) (string-length bare-version)))
+         (number (string->number last-component)))
+    (format "~a~d" prefix (+ number 1))))
```

Comment:
this increment should happen in next-development-version; next i think-release-version should simply drop the -DEV suffix